plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/InlineMethodConverter.kt

```diff
@@ -5,26 +5,45 @@
 
 package org.jetbrains.kotlin.formver.conversion
 
+import org.jetbrains.kotlin.fir.expressions.FirLambdaArgumentExpression
 import org.jetbrains.kotlin.formver.embeddings.MethodSignatureEmbedding
 import org.jetbrains.kotlin.formver.embeddings.VariableEmbedding
 import org.jetbrains.kotlin.formver.viper.MangledName
 import org.jetbrains.kotlin.formver.viper.ast.Exp
 import org.jetbrains.kotlin.formver.viper.ast.Label
 
+sealed interface SubstitutionItem
+data class SubstitutionName(val name: MangledName) : SubstitutionItem
+data class SubstitutionLambda(val lambda: FirLambdaArgumentExpression) : SubstitutionItem
+
 class InlineMethodConverter(
     private val programCtx: ProgramConversionContext,
     override val signature: MethodSignatureEmbedding,
     override val returnVar: VariableEmbedding,
-    private val substitutionParams: Map<MangledName, MangledName>,
+    private val substitutionParams: Map<MangledName, SubstitutionItem>,
+    private val inlineNameAsDefault: Boolean = true // if false, names not in substitutionParams InlineName prefix is not added
```

Comment:
Why is this parameter necessary?

Reply:
When I am in an inline context, a local variable of the inline function is renamed like `inline$pkg$$global$invoke$local$x`. But if a variable is captured by a lambda and the lambda is passed to the inline function, i don't want to rename it. Btw your comment highlighted another problem: variable declared inside the lambda should be renamed and this doesn't happen now. I need to think a bit about this problem